cpp/include/raft/core/mdspan.hpp

```diff
@@ -304,24 +304,53 @@ RAFT_INLINE_FUNCTION auto unravel_index(Idx idx,
   }
 }
 
+/**
+ * @brief Const accessor specialization for default_accessor
+ * 
+ * @tparam ElementType 
+ * @param a 
+ * @return std::experimental::default_accessor<std::add_const_t<ElementType>> 
+ */
+template<class ElementType>
+std::experimental::default_accessor<std::add_const_t<ElementType>>
+accessor_of_const(std::experimental::default_accessor<ElementType> a)
+{
+  return {a};
+}
+
+/**
+ * @brief Const accessor specialization for host_device_accessor
+ * 
+ * @tparam ElementType the data type of the mdspan elements
+ * @tparam MemType the type of memory where the elements are stored.
+ * @param a host_device_accessor
+ * @return host_device_accessor<std::experimental::default_accessor<std::add_const_t<ElementType>>, MemType> 
+ */
+template<class ElementType, memory_type MemType>
+host_device_accessor<std::experimental::default_accessor<std::add_const_t<ElementType>>, MemType>
+accessor_of_const(host_device_accessor<std::experimental::default_accessor<ElementType>, MemType> a)
+{
+  return {a};
+}
+
 /**
  * @brief Create a copy of the given mdspan with const element type
- * @tparam mdspan_type Expected type raft::host_mdspan or raft::device_mdspan
- * @param mds raft::host_mdspan or raft::device_mdspan object
- * @return raft::host_mdspan or raft::device_mdspan with vector_extent
- *         depending on AccessoryPolicy
+ *
+ * @tparam ElementType the data type of the mdspan elements
```

Comment:
Optionally, would you consider the following small documentation change?

```suggestion
 * @tparam ElementType the const-qualified data type of the mdspan elements
```

"The data type of the mdspan elements" is `mdspan::value_type`.  The "element type" (`mdspan::element_type`) is the cv-qualified data type.  The "v" part of "cv qualified" isn't likely to be relevant, so I didn't mention it above.